crates/biome_js_analyze/src/lint/nursery/no_useless_concat.rs

```diff
@@ -0,0 +1,434 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, Rule, RuleDiagnostic, RuleSource,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make::{
+    js_binary_expression, js_string_literal, js_string_literal_expression,
+};
+use biome_js_syntax::{
+    AnyJsExpression, AnyJsLiteralExpression, JsBinaryExpression, JsBinaryOperator,
+};
+use biome_rowan::{AstNode, BatchMutationExt, TextRange, TextSize};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Disallow unnecessary concatenation of literals or template literals.
+    ///
+    /// This rule aims to flag the concatenation of 2 literals when they could be combined into a single literal. Literals can be strings or template literals.
+    /// Concatenation of multiple strings in different lines to prevent big line widths are allowed.
```

Comment:
```suggestion
    /// Concatenation of multiple strings is allowed when the strings are spread over multiple lines in order to prevent exceeding the maximum line width.
```

Reply:
that's way better, done!